src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -99,17 +104,43 @@ public boolean validCheckForObject(final AtlasObject object)
         // Create a ComplexRoundabout based on object
         final ComplexRoundabout complexRoundabout = new ComplexRoundabout((Edge) object,
                 this.leftDrivingCountries);
-        // If the ComplexRoundbaout is invalid add the reasons why to the instructions, so that it
-        // will be flagged
-        if (!complexRoundabout.isValid())
-        {
-            instructions.addAll(complexRoundabout.getAllInvalidations().stream()
-                    .map(ComplexEntity.ComplexEntityError::getReason).collect(Collectors.toSet()));
-        }
         // Get the roundabout Edges
         final Set<Edge> roundaboutEdgeSet = complexRoundabout.getRoundaboutEdgeSet();
         // Get the roundabout Route
         final Route roundaboutEdges = complexRoundabout.getRoundaboutRoute();
+
+        // If the ComplexRoundabout is invalid add the reasons why to the instructions, so that it
+        // will be flagged
+        if (!complexRoundabout.isValid())
+        {
+            // AutoFix candidate only for wrong direction case.
+            if (complexRoundabout.getAllInvalidations().size() == 1 && complexRoundabout
+                    .getAllInvalidations().get(0).getReason().equals(WRONG_WAY_INVALIDATION))
+            {
+                // Mark that the Edges have been processed
+                roundaboutEdgeSet.forEach(
+                        roundaboutEdge -> this.markAsFlagged(roundaboutEdge.getIdentifier()));
+
+                return Optional.of(this
+                        .createFlag(new OsmWayWalker((Edge) object).collectEdges(),
```

Comment:
A roundabout can be made up of multiple Ways, so to would be better to use the `roundaboutEdgeSet` here.

Reply:
Thanks @Bentleysb ! good suggestion

Reply:
@Bentleysb, I moved back to new OsmWayWalker. 1) since we are dealing with single entity Roundabout only. 2) its sorted by way section id unlike roundaboutEdgeSet.